cluster/aws/util.sh

```diff
@@ -721,14 +723,11 @@ function create-iam-profile {
 
 # Creates the IAM roles (if they do not already exist)
 function ensure-iam-profiles {
-  aws iam get-instance-profile --instance-profile-name ${IAM_PROFILE_MASTER} || {
```

Comment:
Ok, I understand your concern, thanks for explaining.

https://github.com/kubernetes/kubernetes/blob/master/cluster/aws/util.sh#L1319-L1339

You'll see that currently the "checks" applied to resource cleanup consist of two unique values:
*  `CLUSTER_ID`, which is traditionally defined to be `INSTANCE_PREFIX` in all the configs I've seen
* `VPC_ID`, which obtained by looking up the ID for `VPC_NAME` (provided in config)

If I applied the same tagging/ deletion check logic to the IAM roles/profiles, that would be consistent with how everything else works.

Sound like a plan?